oauth2/smart/patient-select.php

```diff
@@ -0,0 +1,173 @@
+                                <?php if ($patient['mname']) : ?>
+                                    <?php echo sprintf("%s %s %s", $patient['fname'], $patient['mname'], $patient['lname']); ?>
+                                <?php else : ?>
+                                    <?php echo sprintf("%s %s", $patient['fname'], $patient['lname']); ?>
+                                <?php endif; ?>
+                            </td>
+                            <td>
+                                <?php echo $patient['DOB']; ?>
+                            </td>
+                            <td>
+                                <?php echo $patient['sex']; ?>
+                            </td>
+                            <td>
+                                <?php echo $patient['email']; ?>
+                            </td>
+                            <td>
+                                <button data-patient-id="<?php echo $patient['uuid']; ?>" class="btn btn-primary patient-btn">Select patient</button>
+                            </td>
+                        </tr>
+                        <?php endforeach; ?>
+                    </tbody>
+                </table>
+                <?php endif; ?>
+            </div>
+        </div>
+
+        <form method="post" name="patientForm" id="patientForm" action="<?php echo $redirect ?>">
+            <input type="hidden" name="csrf_token" value="<?php echo attr(CsrfUtils::collectCsrfToken('oauth2')); ?>" />
+            <input id="patient_id" type="hidden" name="patient_id" value="" />
+            <hr />
+<!--            <div class="row">-->
+<!--                <div class="col-md-12">-->
+<!--                    <div class="btn-group">-->
+<!--                        <button type="submit" name="proceed" value="1" class="btn btn-primary">--><?php //echo xlt("Complete Authorization"); ?><!--</button>-->
+<!--                    </div>-->
+<!--                    <div class="form-check-inline float-right">-->
+<!--                        <input class="form-check-input" type="checkbox" name="persist_login" id="persist_login" value="1">-->
+<!--                        <label for="persist_login" class="form-check-label">--><?php //echo xlt("Remember Me"); ?><!--</label>-->
+<!--                    </div>-->
+<!--                </div>-->
+<!--            </div>-->
+        </form>
+    </div>
+</div>
+<script>
+    (function(window) {
+
+        function choosePatient(evt) {
+            var target = evt.target;
+            var patientId = target.dataset.patientId || undefined;
+            if (!patientId) {
+                console.error("<?php echo xlt("Developer error. Patient id is missing from dataset");?>)");
+                return;
+            }
+            var patientInput = document.getElementById('patient_id');
+            if (!patientInput) {
+                console.error("<?php echo xlt("Developer error missing hidden form element 'selectedPatient'");?>)");
```

Comment:
xls

Reply:
xls is the javascript escape function?

Reply:
actually, that is wrong. that is the "old" function. Use xlj() which will translate/escape in javascript context
(the key here is when using this is to have quotes around it which you have already)
https://github.com/openemr/openemr/blob/master/library/htmlspecialchars.inc.php#L177
(js_escape basically calls js_encode)